### PR TITLE
feat: whitelist containers for which logs should be extracted

### PR DESCRIPTION
**Pull request by dejanzele**
## Pull request description 

In case of sidecar containers, i.e. istio-proxy, logs from main containers get fetched in parallel and log output is messy.
This PR introduces whitelisting of containers.

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]

File: pkg/executor/common.go [hunk 1]
```diff
@@ -618,3 +622,11 @@ func GetPodEventsSummary(ctx context.Context, client kubernetes.Interface, pod *
 
 	return message, nil
 }
+
+func IsWhitelistedContainer(containerName string, id string) bool {
+	isMainContainer := containerName == id
+	isInitContainer := containerName == fmt.Sprintf("%s-init", id)
+	isScraperContainer := containerName == fmt.Sprintf("%s-scraper", id)
+	isLogsContainer := containerName == fmt.Sprintf("%s-logs", id)
+	return isMainContainer || isInitContainer || isScraperContainer || isLogsContainer
```
[the remark this answers is not in the document]
Comment (dejanzele) [reply pkg/executor/common.go:631]: Provide them as envvars?

[the remark this answers is not in the document]
Comment (dejanzele) [reply pkg/executor/common.go:631]:
```
WHITELISTED_CONTAINERS=logs,init,scraper
```

[the remark this answers is not in the document]
Comment (exu) [reply pkg/executor/common.go:631]: Yeah list would be great here in case of custom `images` definition

[the remark this answers is not in the document]
Comment (rangoo94) [reply pkg/executor/common.go:631]: > I would image certain user would want to get logs from scraper, etc to debug what went wrong

This one enables all the containers to be displayed. It's only omitting containers added externally (like Istio sidecars)

**Review by rangoo94 — COMMENTED** [review]
One problem I see, is that if someone will modify their Job Templates, may not be able to see logs (if the container names are changed. WDYT?

**Comment by dejanzele** [comment]
@rangoo94 I didn't see the point of naming the container as the execution ID. I'd honestly name it `test`, `scraper`, `log` and `init`. I'd hardcode that in the template.

**Comment by rangoo94** [comment]
@dejanzele, isn't that configurable in the template atm? I think that the whole Job template may be replaced, so we are not even able to decide which container is "ours" - https://github.com/kubeshop/helm-charts/blob/main/charts/testkube-api/templates/_job-template.yaml.tpl#L23-L73

**Comment by dejanzele** [comment]
@rangoo94 it can, but I think it will break as the API code expects this naming schema

**Pull request merged**